auraescript/src/lib.rs

```diff
@@ -82,14 +91,64 @@ mod discovery;
 mod health;
 mod observe;
 
-pub fn init() -> JsRuntime {
+fn get_error_class_name(e: &AnyError) -> &'static str {
+    deno_runtime::errors::get_error_class_name(e).unwrap_or("Error")
+}
+
+pub fn init(main_module: Url) -> MainWorker {
     let extension = Extension::builder("").ops(stdlib()).build();
 
-    JsRuntime::new(RuntimeOptions {
-        extensions: vec![extension],
-        module_loader: Some(Rc::new(TypescriptModuleLoader)),
-        ..Default::default()
-    })
+    let create_web_worker_cb = Arc::new(|_| {
+        todo!("Web workers are not supported yet");
+    });
+    let web_worker_event_cb = Arc::new(|_| {
+        todo!("Web workers are not supported yet");
+    });
+
+    MainWorker::bootstrap_from_options(
+        main_module,
+        PermissionsContainer::allow_all(),
```

Comment:
I remember I said in the discord that I think all permissions are ok, but just so I get double checked on this...

Auraescript will have all [permissions](https://deno.land/manual@v1.32.0/basics/permissions) including:
- get/set env vars
- access to APIs that provide information about user's operating system
- high-resolution time measurement
- network access
- loading of dynamic libraries
- file system read access
- running subprocesses
- file system write access

@dmah42 @krisnova

Reply:
i think your first statement is correct.  as it is run by (and likely written by) a client and can be readily inspected by the client, i think this is fine.

it's like any scripting language: the right function call can expose information or write to the file system, but it's on the person running the script to understand what they're running.

Reply:
I would be inclined to let the client user set and manage these permissions independently of the project. I would honestly just turn the conversation around and have AuraeScript very clearly (AKA "loudly") advertise (EG: In our docs, at the CLI, etc) the permissions it requires to run.

For example we will likely need more detail around this statement:

> access to APIs that provide information about user's operating system

Will AuraeScript be able to mutate kernel parameters on the client's machine? If so? Which ones? Which Linux Capabilities will be required to access a socket on localhost, and so forth.

If we can begin the conversation witih at least documenting these I am fine with it. We can drill into the specifics of each of these later.